components/suggest/src/rs.rs

```diff
@@ -162,6 +162,10 @@ pub(crate) struct DownloadedSuggestionCommonDetails {
 
 /// An AMP suggestion to ingest from an AMP-Wikipedia attachment.
 #[derive(Clone, Debug, Deserialize)]
+// We use deny_unknown_fields for all attachment types so that unrecognized fields result in
+// `UnparsableRecord` entries.  This means that if we add new attachment fields, older clients will
+// ignore the suggestions until they know how to parse them.
+#[serde(deny_unknown_fields)]
```

Comment:
Hmmm, I think this is going a bit too far the other way—it means that older clients will stop being able to display suggestions that they previously could, as soon as we add new fields to them.

I wonder if we can use `#[serde(other)]` to soak up any unknown fields, then check if the map is empty. If it is, we know we can understand the suggestion completely; if it's not, we store the fields we do know, and remember the ID to refetch after a schema version upgrade.

The high-level helpers you added in #6106 would need to be taught about this new case—a record could now be in `unparsable_records` and in `suggestions.record_id` if we understood some of its fields—which is making me wonder if we actually want to track these record IDs in a separate `meta` key, because they're "partially parsable", not "unparsable".

WDYT?

Reply:
I think that makes a lot of sense.  I updated the PR to follow those semantics.

The one thing I didn't do was create a separate meta key for this.  I think it might be simpler to keep 1 key and to change the naming from "unparsable records" to "records to retry" or something like that.  Do we need to differentiate between the two cases?  If so, maybe we could still keep one key but add a field that distinguishes between the two.

Adding another meta key also seems fine to me, I just wanted to discuss it more with you before committing to anything.